ttnn/cpp/ttnn/operations/sliding_window/halo/device/untilize_with_halo_v2_program_factory.cpp

```diff
@@ -33,20 +33,287 @@ struct CBIndices {
     uint32_t out_cb_id = 32;
 
     // Additional CBs for sharded data kernel configs
+    uint32_t padding_config = 32;
+    uint32_t gather_config0 = 32;
+    uint32_t gather_config1 = 32;
+    uint32_t untilize_out_cb_id0 = 32;
+    uint32_t untilize_out_cb_id1 = 32;
+    uint32_t get_next_cb_id() { return next_cb_id++; }
+
+private:
+    uint32_t next_cb_id = tt::CBIndex::c_0;
+};
+
+static inline CBHandle create_circular_buffer(
```

Comment:
for conv2d we started using  tt::tt_metal::create_cb as util maybe you can that as well here instead of rolling your own